pkg/helper/object_meta_merger.go

```diff
@@ -39,4 +39,27 @@ func MergeMapStringString(modified *bool, existing *map[string]string, desired m
 			*modified = true
 		}
 	}
+
+}
+
+func MergeMapStringStringLabels(modified *bool, existing *map[string]string, desired map[string]string) {
+	if *existing == nil {
+		*existing = map[string]string{}
+	}
+
+	for k, v := range desired {
+		if existingVal, ok := (*existing)[k]; !ok || v != existingVal {
+			(*existing)[k] = v
+			*modified = true
+		}
+	}
+
+	// Remove existing values that are not desired
+	for k := range *existing {
+		if _, ok := desired[k]; !ok {
+			delete(*existing, k)
+			*modified = true
+		}
+	}
+
```

Comment:
@Patryk-Stefanski , I'd like ask why not using existing function MergeMapStringString(...) ?

Reply:
the existing function doesn't remove labels that are not desired, this meant that removing a label in the APIManagerCR wouldn't remove it from the component.

Reply:
But then this implementation is not allowing to add any arbitrary label to the pod template labels. And I think that the operator should allow that.

Reply:
Sorry not quite sure what you mean here. This new implementation still allows for the creation of new labels for each deploymentConfig/Pod and ensures that whatever labels are defined in the APIManager CR are correctly reflected on a pod. Without this logic, deleted labels in the APIManager CR will be still present on the pod or is this the expected behaviour?

Reply:
Hmm, this would mean that customers would have to manually delete the labels from the deployment config to remove outdated labels, not sure if this is the best approach. Any ideas of how we could get the best of both?

Reply:
nothing robust. All I can think about could easily be broken. 

I am just thinking loudly: What if we define an annotation for the APIManager CRD and if the operator finds that, it will play with hard constraints (regarding labels and annotations comming). By default, it will be permissive regarding existing labels. (not saying it should be implemented in this PR)

Reply:
Yeah, I like the idea of using an annotation to define the expected behvaviour on the reconcilation of labels and annotations to pod templates 👍

Reply:
Great I changed the logic back and let's discuss how to proceed with the annotations/labels functionality on Wednesday's sync call.